### PR TITLE
Make Auth SDK simpler to use

### PR DESCRIPTION
**Pull request by zaiste**
## Breaking Change

- [x] removed the `useSaleorAuthClient` hook in favor of the `createSaleorAuthClient` function
- [x] removed the hook wrapper for Apollo client
- [x] removed the hook wrapper for urql client

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -139,38 +147,39 @@ Let's look at an example:
...
-    fetch: useSaleorAuthClientProps.saleorAuthClient.fetchWithAuth,
+const makeUrqlClient = () =>
+  createClient({
+    url: saleorApiUrl,
+    fetch: saleorAuthClient.fetchWithAuth,
     exchanges: [cacheExchange, fetchExchange],
   });
 
+export default function App({ Component, pageProps }: AppProps) {
+  const [urqlClient, setUrqlClient] = useState<Client>(makeUrqlClient());
```
Comment (mmiszy) [inline README.md:164]: Maybe we should add a comment with a link to the issue that describes this pattern?
```suggestion
  // https://github.com/urql-graphql/urql/issues/297#issuecomment-504782794
  const [urqlClient, setUrqlClient] = useState<Client>(makeUrqlClient());
```

File: src/react/SaleorAuthProvider.tsx [hunk 2]
```diff
@@ -1,19 +1,12 @@
...
-  children,
-  saleorAuthClient,
-  isAuthenticating,
-}: PropsWithChildren<UseSaleorAuthClient>) => {
-  invariant(
-    saleorAuthClient,
-    "Missing Saleor Auth Client - are you sure you created it using useSaleorAuthClient?",
-  );
+export const SaleorAuthProvider = ({ children, client }: PropsWithChildren<{ client: SaleorAuthClient }>) => {
+  invariant(client, "Missing Saleor Auth Client - are you sure you created it using useSaleorAuthClient?");
```
Comment (mmiszy) [inline src/react/SaleorAuthProvider.tsx:7]: Is it possible that `client` is falsy here? Why?

**Review by mmiszy — CHANGES_REQUESTED** [review]
We're missing a BREAKING CHANGE warning in the description of the PR.

File: README.md [hunk 3]
```diff
@@ -213,7 +241,7 @@ export const LoginPage = () => {
   }
 
   return (
-    <section>
+    <main className={`flex min-h-screen flex-col items-center justify-between p-24`}>
```
Comment (mmiszy) [inline README.md:245]: We should skip `className`; it's unrelated to auth and only obscures what we're trying to teach here.

**Review by mmiszy — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (zaiste) [reply README.md:245]: @mmiszy it looks pretty bad without, but OK; could be confusing long term. I have another solution for that later

**Review by mmiszy — APPROVED** [review]

**Pull request merged**